subprojects/test-runner/service/src/main/kotlin/com/avito/runner/service/DeviceWorkerPoolImpl.kt

```diff
@@ -18,33 +18,31 @@ import kotlinx.coroutines.CoroutineName
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.channels.Channel
 import kotlinx.coroutines.channels.ReceiveChannel
+import kotlinx.coroutines.channels.SendChannel
 import kotlinx.coroutines.launch
 import java.io.File
 
-class IntentionExecutionServiceImplementation(
+class DeviceWorkerPoolImpl(
     private val outputDirectory: File,
     private val loggerFactory: LoggerFactory,
     private val devices: ReceiveChannel<Device>,
     private val intentionsRouter: IntentionsRouter = IntentionsRouter(loggerFactory = loggerFactory),
     private val testListener: TestListener,
     private val deviceMetricsListener: DeviceListener,
     private val timeProvider: TimeProvider,
+    private val intentions: ReceiveChannel<Intention>,
+    private val intentionResults: SendChannel<IntentionResult>,
+    private val deviceSignals: SendChannel<Device.Signal>,
```

Comment:
just an idea: maybe extract channels to `DviceWorkerPoolState` class, like `TestRunnerExecutionState`?